python/cudf/cudf/utils/ioutils.py

```diff
@@ -444,6 +492,20 @@
     A list of column names which should be written as map type in the ORC file.
     Note that this option only affects columns of ListDtype. Names of other
     column types will be ignored.
+storage_options : dict, optional, default None
+    Extra options that make sense for a particular storage connection,
+    e.g. host, port, username, password, etc. For HTTP(S) URLs the key-value
+    pairs are forwarded to ``urllib.request.Request`` as header options.
+    For other URLs (e.g. starting with "s3://", and "gcs://") the key-value
+    pairs are forwarded to ``fsspec.open``. Please see ``fsspec`` and
+    ``urllib`` for more details.
+index : bool, default None
+    If ``True``, include the dataframe's index(es) in the file output. If
+    ``False``, they will not be written to the file. If ``None``, the
+    engine's default behavior will be used. However, instead of being saved
+    as values, the ``RangeIndex`` will be stored as a range in the metadata
+    so it doesn’t require much space and is faster. Other indexes will
```

Comment:
It's somewhat unclear under what circumstances `RangeIndex` is stored in compact form. I think:

- `True` => `RangeIndex` expanded, all indices written
- `False` => No indices written
- `None` => `RangeIndex` compact, all indices written

But this wording leaves open the idea that the engine default might be to not write any indices. Is this intentional?

Reply:
> I think:
> 
> * `True` => `RangeIndex` expanded, all indices written
> * `False` => No indices written
> * `None` => `RangeIndex` compact, all indices written

This is correct and that is the intended behavior. However, the docstring has been copied from [pandas](https://pandas.pydata.org/pandas-docs/version/1.1/reference/api/pandas.DataFrame.to_parquet.html), let me know if you have a suggestion to improve the wording here we could definitely do that here.

Reply:
The pandas version is:

> If True, include the dataframe’s index(es) in the file output. If False, they will not be written to the file. _If None, similar to True the dataframe’s index(es) will be saved._ However, instead of being saved as values, the RangeIndex will be stored as a range in the metadata so it doesn’t require much space and is faster. Other indexes will be included as columns in the file output.

So no talk of "the engine's default". Which I think would clarify things. I would also write:

", however, instead of being saved as values, _any_ RangeIndex will be stored ..."

Two changes: "however" follows on from previous sentence, "any" rather than "the" `RangeIndex`

Reply:
Concretely:

```
     If ``True``, include the dataframe's index(es) in the file output. If
     ``False``, they will not be written to the file. If ``None``, similar to ``True`` 
     the dataframe's index(es) will be saved, however, instead of being saved
     as values any ``RangeIndex`` will be stored as a range in the metadata
     so it doesn’t require much space and is faster.
```

Reply:
Sounds good to me 👍 Updated the docstring.